### PR TITLE
Port config poller updates

### PR DESCRIPTION
**Pull request by connorwstein**
- Simplify the contract polling for OCR2 by using the log poller
- Performance improvement in the LatestTransmission details via log poller
- Finally time to introduce a log poller mock
- Misc: support from address for eth calls using the backend sim wrapper, a handful of new utils

### CODE REVIEW

File: core/chains/evm/logpoller/log_poller.go [hunk 1]
```diff
@@ -416,56 +439,56 @@ func (lp *LogPoller) findLCA(h common.Hash) (int64, error) {
...
+func (lp *logPoller) IndexedLogsTopicGreaterThan(eventSig common.Hash, address common.Address, topicIndex int, topicValueMin common.Hash, confs int, qopts ...pg.QOpt) ([]Log, error) {
 	return lp.orm.SelectIndexLogsTopicGreaterThan(address, eventSig[:], topicIndex, topicValueMin, confs, qopts...)
 }
 
-func (lp *LogPoller) IndexedLogsTopicRange(eventSig common.Hash, address common.Address, topicIndex int, topicValueMin common.Hash, topicValueMax common.Hash, confs int, qopts ...pg.QOpt) ([]Log, error) {
+func (lp *logPoller) IndexedLogsTopicRange(eventSig common.Hash, address common.Address, topicIndex int, topicValueMin common.Hash, topicValueMax common.Hash, confs int, qopts ...pg.QOpt) ([]Log, error) {
 	return lp.orm.SelectIndexLogsTopicRange(address, eventSig[:], topicIndex, topicValueMin, topicValueMax, confs, qopts...)
 }
 
+func (lp *logPoller) LatestBlock(qopts ...pg.QOpt) (int64, error) {
```
Comment (connorwstein) [inline core/chains/evm/logpoller/log_poller.go:471]: this shuffle just aligns with other repos to keep updates simpler

File: core/utils/utils.go [hunk 2]
```diff
@@ -37,6 +40,36 @@ const (
...
+}
+
+func RandomBytes32() (r [32]byte) {
+	b := make([]byte, 32)
+	_, _ = rand.Read(b[:]) // Assignment for errcheck. Only used in tests so we can ignore.
+	copy(r[:], b)
+	return
+}
+
+func Bytes32ToSlice(a [32]byte) (r []byte) {
```
[the remark this answers is not in the document]
Comment (connorwstein) [reply core/utils/utils.go:56]: ditto

File: core/chains/evm/logpoller/log_poller.go [hunk 3]
```diff
@@ -114,8 +137,8 @@ func (lp *LogPoller) Replay(ctx context.Context, fromBlock int64) error {
 	return nil
 }
 
-func (lp *LogPoller) Start(parentCtx context.Context) error {
-	return lp.StartOnce("LogPoller", func() error {
+func (lp *logPoller) Start(parentCtx context.Context) error {
+	return lp.StartOnce("logPoller", func() error {
```
Comment (jmank88) [inline core/chains/evm/logpoller/log_poller.go:141]: nit/ were these strings caught in a replace-all for the type?

File: core/services/ocrcommon/addressparser.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
+package ocrcommon
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
```
Comment (jmank88) [inline core/services/ocrcommon/addressparser.go:7]: nit/ Would it make sense to isolate evm stuff in a sub-package?

File: core/utils/ethabi.go [hunk 5]
```diff
@@ -232,6 +232,11 @@ func Bytes4FromString(s string) [4]byte {
 	return b4
 }
 
+func MustAbiType(ts string, components []abi.ArgumentMarshaling) abi.Type {
+	ty, _ := abi.NewType(ts, "", components)
```
Comment (jmank88) [inline core/utils/ethabi.go:236]: Would it be better to panic on error than silently ignore?

File: core/services/ocrcommon/addressparser.go (hunk 4 above)
Comment (connorwstein) [reply core/services/ocrcommon/addressparser.go:7]: oh yeah actually this is better housed just in the relay/evm package

**Review by jmank88 — APPROVED** [review]

**Pull request merged**